core/types/index.d.ts

```diff
@@ -399,22 +400,28 @@ declare module '@verdaccio/types' {
     search(onPackage: onSearchPackage, onEnd: onEndSearchPackage, validateName: onValidatePackage): void;
   }
 
+  type StorageUpdateCallback = (data: Package, cb: CallbackAction) => void;
+  type StorageUpdateHandler = (name: string, cb: StorageUpdateCallback) => void;
+  type StorageWriteCallback = (name: string, json: Package, callback: Callback) => void;
+  type PackageTransformer = (pkg: Package) => Package;
+  type ReadPackageCallback = (err: any | null, data?: Package) => void;
```

Comment:
`err: any `

how about `err: Error | null` ?

Reply:
Verdccio core not use `Error`, it uses `VerdaccioError` which are http status codes, the problem is these types are in `@verdaccio/common-api` and include them here would create a circle dependency. I added a comment in that file so in the future we can find a better solution, for now it must be `any`.

Reply:
Plugins must return errors defined in `@verdaccio/common-api `, there are a complete library of methods ready to use, this will be documented in future blog posts.

Reply:
https://github.com/verdaccio/monorepo/blob/f0887bdc71a239e013fb4142d5e23e3cae9068b7/core/types/index.d.ts#L385-L386

This one, I see !!!
will it break if you can move that here and use this package in `common-api` ? for types ?

Reply:
No, but `@verdaccio/types` must be free of dependencies, and `VerdaccioError` has one from `http-status` ... I think we must create a custom type and get rid of `http-status in the future`, so then, that would not be a problem anymore.

Reply:
> I think we must create a custom type and get rid of http-status in the future, so then, that would not be a problem anymore.

Yup this one is better solution I think

Reply:
🙃 so ... can I merge?

Reply:
sure 😀